### PR TITLE
Remove deprecated Monitor class and its dependencies. Resolves #2367, resolves #2572

### LINKED ISSUE #2367: Remove monitor wrapper and associated code entirely
Per numerous issues and PRs, the monitor wrapper is now fully depricated, with deprecation warning. Accordingly, it should be removed in the future. My vote is doing this some time in January 2022

### LINKED ISSUE #2572: Potential dead code in utils
Without any further review on my end, the following files possibly are dead code, and/or should be removed for refactoring purposes:

https://github.com/openai/gym/blob/master/gym/utils/atomic_write.py
https://github.com/openai/gym/blob/master/gym/utils/closer.py
https://github.com/openai/gym/blob/master/gym/utils/json_utils.py

### PR DESCRIPTION
**Pull request by gianlucadecola**
This PR removes the deprecated Monitor class as well as the StatsRecorder class (which is used only by Monitor). As a consequence, the following files are dead-code not needed anymore:

https://github.com/openai/gym/blob/master/gym/utils/atomic_write.py
https://github.com/openai/gym/blob/master/gym/utils/closer.py
https://github.com/openai/gym/blob/master/gym/utils/json_utils.py

### CODE REVIEW

**Comment by jkterry1** [comment]
@gianlucadecola could you please fix tests?

File: gym/utils/closer.py [hunk 1]
```diff
@@ -1,68 +0,0 @@
-import atexit
-import threading
-import weakref
-
-
-class Closer:
```
Comment (vwxyzjn) [inline gym/utils/closer.py:6]: I see you have removed `Closer`, which I recall is used in some cases such as in #2316. @XuehaiPan would you mind reviewing this?

**Review by vwxyzjn — COMMENTED** [review]
This is awesome! Looking forward to completely remove `Monitor` class.

**Comment by gianlucadecola** [comment]
> @gianlucadecola could you please fix tests?

Fixed :)

**Comment by gianlucadecola** [comment]
Fixes a little conflict in car_racing.py and rebased

**Comment by jkterry1** [comment]
@vwxyzjn are your changes made?

**Comment by vwxyzjn** [comment]
I am happy with the changes as is, but I thought maybe @XuehaiPan would want to keep the `Closer` object.

File: gym/utils/closer.py (hunk 1 above)
Comment (XuehaiPan) [reply gym/utils/closer.py:6]: That's fine. The legacy `Closer` can be implement with `__del__`:

```
class MyClass:
    def close(self):
        ...
    
    def __del__(self):
        self.close()
```

**Pull request merged**